frappe/integrations/doctype/oauth_provider_settings/oauth_provider_settings.py

```diff
@@ -16,12 +27,112 @@ class OAuthProviderSettings(Document):
 		from frappe.types import DF
 
 		skip_authorization: DF.Literal["Force", "Auto"]
+		jwks_enabled: DF.Check
+		jwks_key_id: DF.Data
+		jwks_private_key: DF.Code
+		jwks_public_key: DF.Code
+		jwks_expiry: DF.Datetime
 	# end: auto-generated types
-	pass
+	
+	def validate(self):
+		if self.jwks_enabled and not self.jwks_private_key:
+			self.generate_keys()
+			
+	def generate_keys(self):
+		"""Generate a new RSA key pair for JWKS"""
+		# Generate private key
+		private_key = rsa.generate_private_key(
+			public_exponent=65537,
+			key_size=2048,
+			backend=default_backend()
+		)
+		
+		# Get private key in PEM format
+		private_pem = private_key.private_bytes(
+			encoding=serialization.Encoding.PEM,
+			format=serialization.PrivateFormat.PKCS8,
+			encryption_algorithm=serialization.NoEncryption()
+		).decode('utf-8')
+		
+		# Get public key in PEM format
+		public_key = private_key.public_key()
+		public_pem = public_key.public_bytes(
+			encoding=serialization.Encoding.PEM,
+			format=serialization.PublicFormat.SubjectPublicKeyInfo
+		).decode('utf-8')
+		
+		# Generate key ID
+		kid = str(uuid.uuid4())
+		
+		# Set expiry date (30 days from now by default)
+		expiry = add_days(now_datetime(), 30)
+		
+		# Save the key information
+		self.jwks_private_key = private_pem
+		self.jwks_public_key = public_pem
+		self.jwks_key_id = kid
+		self.jwks_expiry = expiry
+		
+	def get_jwks(self):
+		"""Return the JWKS in the standard format"""
+		if not self.jwks_enabled or not self.jwks_public_key:
+			return {"keys": []}
+			
+		# Parse the public key to get components
+		public_key = serialization.load_pem_public_key(
+			self.jwks_public_key.encode('utf-8'),
+			backend=default_backend()
+		)
+		
+		# Get the key components
+		numbers = public_key.public_numbers()
+		
+		# Create the JWK
+		jwk = {
+			"kty": "RSA",
+			"use": "sig",
+			"kid": self.jwks_key_id,
+			"n": base64url_encode(numbers.n.to_bytes((numbers.n.bit_length() + 7) // 8, byteorder='big')).decode('utf-8'),
+			"e": base64url_encode(numbers.e.to_bytes((numbers.e.bit_length() + 7) // 8, byteorder='big')).decode('utf-8'),
+			"alg": "RS256",
+		}
+		
+		return {"keys": [jwk]}
```

Comment:
here there should be at least 2 keys, that makes rotation seamless.

that means on client side, at least 1 kid is there from the available keys.